pkg/secretless/config/v2/service.go

```diff
@@ -186,6 +188,60 @@ func connectorID(
 	return connectorID, nil
 }
 
+// ValidateConfig is a function object that validates the connectors requested
+// against the plugins available. Its static dependencies are a logger and the
+// plugins available to create connectors.
+type ValidateConfig struct {
+	logger logapi.Logger
+	availPlugins plugin.AvailablePlugins
+}
+
+// NewValidateConfig creates a new instance of ValidateConfig.
+func NewValidateConfig(lg logapi.Logger, ap plugin.AvailablePlugins) ValidateConfig {
+	return ValidateConfig{
+		logger: lg,
```

Comment:
Inconsistent variable naming - `l` should be fine here if you're using golang naming standard or you can use `logger` to match some of our other code.